src/pages/AuthenticationPage/AuthenticationComp/AuthSignUp/AuthSignUp.jsx

```diff
@@ -1,113 +1,179 @@
-        className="signup-form"
-      >
-        <h2 className="form-title txt-center h2 ">Create Account</h2>
-        <div className="flex-container social-login">
-          <a href="#">
-            <img
-              className="icon size-xs"
-              src="https://img.icons8.com/color/48/000000/google-logo.png"
-            />
-          </a>
-          <a href="#">
-            <img
-              className="icon size-xs"
-              src="https://img.icons8.com/ios-glyphs/30/000000/github.png"
-            />
-          </a>
-          <a href="#">
-            <img
-              className="icon size-xs"
-              src="https://img.icons8.com/color/48/000000/twitter--v1.png"
-            />
-          </a>
-        </div>
-        <p className="or">
-          <span>or</span>
-        </p>
-        <div className="name-input-box">
-          <div className="input-with-icon ">
-        < FaUserAlt className="icon size-xs" />
+    if (!passwordExpression.test(inputState.password) || !inputState.password2.match(passwordExpression)){
+      console.log(!passwordExpression.test(inputState.password),'1');
+      console.log((inputState.password2.match(passwordExpression)),"2");
+      console.log("password should follow rules");
+      return false
+    }
+    return true;
+  };
+
+  
+
+  function submitFormHandler(e) {
+    e.preventDefault();
+      signUpUser(inputState.firstName,inputState.lastName,inputState.email,inputState.password,setformSubmitState);;
+  }
+  return (
+    <>
+      <Navbar isMenuRequired={false} />
+      <section className="flex form-container">
+        <form
+          onSubmit={submitFormHandler}
+          action="sign-up form register"
+          className="signup-form"
+        >
+          <h2 className="form-title txt-center h2 ">Create Account</h2>
+          <div className="flex-container social-login">
+            <a href="#">
+              <img
+                className="icon size-xs"
+                src="https://img.icons8.com/color/48/000000/google-logo.png"
+              />
+            </a>
+            <a href="#">
+              <img
+                className="icon size-xs"
+                src="https://img.icons8.com/ios-glyphs/30/000000/github.png"
+              />
+            </a>
+            <a href="#">
+              <img
+                className="icon size-xs"
+                src="https://img.icons8.com/color/48/000000/twitter--v1.png"
+              />
+            </a>
+          </div>
+          <p className="or">
+            <span>or</span>
+          </p>
+          <div className="name-input-box">
+            <div className="input-with-icon ">
+              <FaUserAlt className="icon size-xs" />
+              <input
+                type="text"
+                required
+                onChange={inputUpdate}
+                placeholder="First Name"
+                name="firstName"
+                className="input firstname-input"
```

Comment:
suggestion - you can add the value attribute & assign it with the state variable to make it a controlled component

Reply:
I actually want to read more about controlled and uncontrolled input and then will implement 1.